src/main.rs

```diff
@@ -129,116 +130,104 @@ fn metadata(path: &Path) -> Result<Metadata, Error> {
 }
 
 fn main() -> anyhow::Result<()> {
-    let matches = cli::arg_matches();
-
-    if let Some(matches) = matches.subcommand_matches("sweep") {
-        let verbose = matches.is_present("verbose");
-        setup_logging(verbose);
-
-        let dry_run = matches.is_present("dry-run");
-
-        // Default to current invocation path.
-        let path = match matches.value_of("path") {
-            Some(p) => PathBuf::from(p),
-            None => env::current_dir().expect("Failed to get current directory"),
-        };
+    let args = cli::parse();
+
+    let criterion = args.criterion();
+    let dry_run = args.dry_run;
+    setup_logging(args.verbose);
+
+    // Default to current invocation path.
+    let path = args
+        .path
+        .unwrap_or_else(|| env::current_dir().expect("Failed to get current directory"));
+
+    if let Criterion::Stamp = criterion {
+        debug!("Writing timestamp file in: {:?}", path);
+        return Timestamp::new()
+            .store(path.as_path())
+            .context("Failed to write timestamp file");
+    }
 
-        if matches.is_present("stamp") {
-            debug!("Writing timestamp file in: {:?}", path);
-            return Timestamp::new()
-                .store(path.as_path())
-                .context("Failed to write timestamp file");
+    let paths = if args.recursive {
+        find_cargo_projects(&path, args.hidden)
+    } else {
+        let metadata = metadata(&path).context(format!(
+            "Failed to gather metadata for {:?}",
+            path.display()
+        ))?;
+        let out = Path::new(&metadata.target_directory).to_path_buf();
+        if out.exists() {
+            vec![out]
+        } else {
+            anyhow::bail!("Failed to clean {:?} as it does not exist.", out);
         }
+    };
 
-        let paths = if matches.is_present("recursive") {
-            find_cargo_projects(&path, matches.is_present("hidden"))
-        } else {
-            let metadata = metadata(&path).context(format!(
-                "Failed to gather metadata for {:?}",
-                path.display()
-            ))?;
-            let out = Path::new(&metadata.target_directory).to_path_buf();
-            if out.exists() {
-                vec![out]
-            } else {
-                anyhow::bail!("Failed to clean {:?} as it does not exist.", out);
-            }
+    if matches!(criterion, Criterion::Installed | Criterion::Toolchains(_)) {
+        let toolchains = match criterion {
+            Criterion::Installed => vec![],
+            Criterion::Toolchains(vec) => vec,
+            _ => todo!(),
         };
```

Comment:
```suggestion
    let toolchains = match criterion {
        Criterion::Installed => Some(vec![]),
        Criterion::Toolchains(vec) => Some(vec),
        _ => None,
    };
    if let Some(toolchains) = toolchains {

Reply:
I think this is a small problem when reviewing by each commit, seems like you commented on outdated code.